stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -0,0 +1,1056 @@
+    StacksDBTx, StacksHeaderInfo,
+};
+use super::stacks::events::StacksTransactionReceipt;
+use super::stacks::Error as ChainstateError;
+use super::stacks::StacksBlock;
+use super::stacks::StacksTransaction;
+use super::stacks::{StacksBlockHeader, StacksMicroblock, TransactionPayload};
+use crate::burnchains::PoxConstants;
+use crate::chainstate::burn::db::sortdb::SortitionDB;
+use crate::chainstate::stacks::db::StacksChainState;
+use crate::chainstate::stacks::{MINER_BLOCK_CONSENSUS_HASH, MINER_BLOCK_HEADER_HASH};
+use crate::clarity_vm::clarity::{ClarityInstance, PreCommitClarityBlock};
+use crate::clarity_vm::database::SortitionDBRef;
+use crate::monitoring;
+use crate::util_lib::db::{u64_to_sql, Error as DBError, FromRow};
+
+#[cfg(test)]
+pub mod tests;
+
+define_named_enum!(HeaderTypeNames {
+    Nakamoto("nakamoto"),
+    Epoch2("epoch2"),
+});
+
+impl ToSql for HeaderTypeNames {
+    fn to_sql(&self) -> rusqlite::Result<rusqlite::types::ToSqlOutput<'_>> {
+        self.get_name_str().to_sql()
+    }
+}
+
+impl FromSql for HeaderTypeNames {
+    fn column_result(value: rusqlite::types::ValueRef<'_>) -> rusqlite::types::FromSqlResult<Self> {
+        Self::lookup_by_name(value.as_str()?).ok_or_else(|| FromSqlError::InvalidType)
+    }
+}
+
+pub const NAKAMOTO_CHAINSTATE_SCHEMA_1: &'static [&'static str] = &[
+    r#"
+      -- Table for Nakamoto Block Headers
+      CREATE TABLE nakamoto_block_headers (
+          -- The following fields all correspond to entries in the StacksHeaderInfo struct
+                     block_height INTEGER NOT NULL,
+                     -- root hash of the internal, not-consensus-critical MARF that allows us to track chainstate/fork metadata
+                     index_root TEXT NOT NULL,
+                     -- all consensus hashes are guaranteed to be unique
+                     consensus_hash TEXT UNIQUE NOT NULL,
+                     -- burn header hash corresponding to the consensus hash (NOT guaranteed to be unique, since we can 
+                     --    have 2+ blocks per burn block if there's a PoX fork)
+                     burn_header_hash TEXT NOT NULL,
+                     -- height of the burnchain block header that generated this consensus hash
+                     burn_header_height INT NOT NULL,
+                     -- timestamp from burnchain block header that generated this consensus hash
+                     burn_header_timestamp INT NOT NULL,
+                     block_size TEXT NOT NULL,
+          -- The following fields all correspond to entries in the NakamotoBlockHeader struct
+                     version INTEGER NOT NULL,
+                     -- this field is the total number of blocks in the chain history (including this block)
+                     chain_length INTEGER NOT NULL,
+                     -- this field is the total amount of BTC spent in the chain history (including this block)
+                     btc_spent INTEGER NOT NULL,
```

Comment:
Can this be renamed to `burn_spent` to keep it consistent with the burnchain nomenclature in the code?

Reply:
Updated